atom/renderer/api/lib/remote.coffee

```diff
@@ -6,17 +6,23 @@ CallbacksRegistry = require 'callbacks-registry'
 callbacksRegistry = new CallbacksRegistry
 
 # Convert the arguments object into an array of meta data.
-wrapArgs = (args) ->
+wrapArgs = (args, visited=[]) ->
   valueToMeta = (value) ->
     if Array.isArray value
-      type: 'array', value: wrapArgs(value)
+      type: 'array', value: wrapArgs(value, visited)
     else if Buffer.isBuffer value
       type: 'buffer', value: Array::slice.call(value, 0)
     else if value? and typeof value is 'object' and v8Util.getHiddenValue value, 'atomId'
       type: 'remote-object', id: v8Util.getHiddenValue value, 'atomId'
     else if value? and typeof value is 'object'
       ret = type: 'object', name: value.constructor.name, members: []
-      ret.members.push(name: prop, value: valueToMeta(field)) for prop, field of value
+      for prop, field of value
+        if typeof field is 'object'
+          if visited.indexOf(field) != -1
```

Comment:
In CoffeScript people usually use `field in visited` instead of `indexOf != -1`.